docarray/documents/legacy/legacy_document.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from typing import Any, Dict, Optional
+
+from docarray import BaseDocument, DocumentArray
+from docarray.typing import AnyEmbedding, AnyTensor
+
+
+class Document(BaseDocument):
```

Comment:
I think we have to discuss the name. `Document` gives it too much importance, `LegaceDocument` is maybe too much of a "we won't support this anymore" vibe. Maybe `ClassicDocument`?

Reply:
but it is Document only under the `docarray.documents.legacy` namespace. But yeah maybe if someone search in the codebase it will find this Document and think it is important.

I think `LegacyDocument` is not too much. Nothing promise we will support this forever. Lets in the future we have to make a choice on a API that will make this Document obsolete then I am not sure we want to say. "No we cannot do it bc we will break `ClassicDocument`"

Reply:
I think `LegacyDocument` is good enough

Reply:
done